packages/flutter/lib/src/painting/border_radius.dart

```diff
@@ -691,26 +690,26 @@ class BorderRadiusDirectional extends BorderRadiusGeometry {
   /// If either is null, this function interpolates from [BorderRadiusDirectional.zero].
   ///
   /// {@macro dart.ui.shadow.lerp}
-  static BorderRadiusDirectional lerp(BorderRadiusDirectional a, BorderRadiusDirectional b, double t) {
+  static BorderRadiusDirectional? lerp(BorderRadiusDirectional? a, BorderRadiusDirectional? b, double t) {
     assert(t != null);
     if (a == null && b == null)
       return null;
     if (a == null)
-      return b * t;
+      return b! * t;
     if (b == null)
       return a * (1.0 - t);
     return BorderRadiusDirectional.only(
-      topStart: Radius.lerp(a.topStart, b.topStart, t),
-      topEnd: Radius.lerp(a.topEnd, b.topEnd, t),
-      bottomStart: Radius.lerp(a.bottomStart, b.bottomStart, t),
-      bottomEnd: Radius.lerp(a.bottomEnd, b.bottomEnd, t),
+      topStart: Radius.lerp(a.topStart, b.topStart, t)!,
+      topEnd: Radius.lerp(a.topEnd, b.topEnd, t)!,
+      bottomStart: Radius.lerp(a.bottomStart, b.bottomStart, t)!,
+      bottomEnd: Radius.lerp(a.bottomEnd, b.bottomEnd, t)!,
     );
   }
 
   @override
-  BorderRadius resolve(TextDirection direction) {
+  BorderRadius resolve(TextDirection? direction) {
     assert(direction != null);
-    switch (direction) {
+    switch (direction!) {
```

Comment:
Why is this nullable when we assert that it is not null?

Reply:
After trying to make direction non-nullable there are several failure in tests. null is actually possible as argument for `BorderRadius.resolve`. So I'll revert the last commit.